service/src/main/java/uk/nhs/adaptors/gp2gp/gpc/GetGpcDocumentTaskExecutor.java

```diff
@@ -1,21 +1,58 @@
 package uk.nhs.adaptors.gp2gp.gpc;
 
-import org.springframework.stereotype.Component;
+import java.util.UUID;
 
+import lombok.SneakyThrows;
 import lombok.extern.slf4j.Slf4j;
+import uk.nhs.adaptors.gp2gp.common.storage.StorageConnectorService;
+import uk.nhs.adaptors.gp2gp.common.storage.StorageDataWrapper;
 import uk.nhs.adaptors.gp2gp.common.task.TaskExecutor;
+import uk.nhs.adaptors.gp2gp.ehr.EhrExtractStatusService;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
 
 @Slf4j
 @Component
 public class GetGpcDocumentTaskExecutor implements TaskExecutor<GetGpcDocumentTaskDefinition> {
+    private static final String JSON_EXTENSION = ".json";
+
+    @Autowired
+    private StorageConnectorService storageConnectorService;
+    @Autowired
+    private EhrExtractStatusService ehrExtractStatusService;
+    @Autowired
+    private GpcRequestBuilder gpcRequestBuilder;
+    @Autowired
+    private GpcClient gpcClient;
 
     @Override
     public Class<GetGpcDocumentTaskDefinition> getTaskType() {
         return GetGpcDocumentTaskDefinition.class;
     }
 
     @Override
-    public void execute(GetGpcDocumentTaskDefinition taskDefinition) {
+    @SneakyThrows
+    public void execute(GetGpcDocumentTaskDefinition documentTaskDefinition) {
         LOGGER.info("Execute called from GetGpcDocumentTaskExecutor");
+
+        var request = gpcRequestBuilder.buildGetDocumentRecordRequest(documentTaskDefinition);
+        var response = gpcClient.getDocumentRecord(request, documentTaskDefinition);
+
+        String documentName = documentTaskDefinition.getDocumentId() + JSON_EXTENSION;
+        String taskId = UUID.randomUUID().toString();
+        storageConnectorService.uploadDocument(documentName, buildStorageDataWrapper(documentTaskDefinition, response, taskId));
+        ehrExtractStatusService.updateEhrExtractStatusAccessDocument(documentTaskDefinition, documentName, taskId);
+    }
+
+    private StorageDataWrapper buildStorageDataWrapper(GetGpcDocumentTaskDefinition documentTaskDefinition,
```

Comment:
we could change prams to TaskDefinition and move this to a separate class for both structured and document?

Reply:
Done